content/en/blog/_posts/2024-02-29-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,421 @@
+`Unconfined`.
+
+## CRI-O to the rescue
+
+The Kubernetes container runtime [CRI-O](https://github.com/cri-o/cri-o)
+provides various features using custom annotations. The upcoming v1.30 release
+will [add support for a new annotation](https://github.com/cri-o/cri-o/pull/7719)
+called `io.kubernetes.cri-o.seccompProfile`. This annotation allows to specify:
+
+- a seccomp profile for a specific container, when used as:
+  `io.kubernetes.cri-o.seccompProfile/<CONTAINER>`
+- a seccomp profile for every container within a pod, when used without the
+  container name suffix:
+  `io.kubernetes.cri-o.seccompProfile`
+- a seccomp profile for a whole container image, if the image itself contains
+  the annotation `io.kubernetes.cri-o.seccompProfile`
+
+CRI-O will only respect the annotation if the runtime is configured to allow it,
+as well as for workloads running as `Unconfined`. All other workloads will still
+use the value from the `securityContext` with a higher priority.
+
+The annotation alone will not help much with the distribution of the profiles,
+but the way how they can be referenced will! For example, users can now specify
+seccomp profiles like regular container images by using OCI artifacts:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    io.kubernetes.cri-o.seccompProfile: quay.io/crio/seccomp:v1
+spec: …
+```
+
+The image `quay.io/crio/seccomp:v1` contains a `seccomp.tar` tarball, which
+contains the actual profile. Tools like [ORAS](https://oras.land) or
+[Skopeo](https://github.com/containers/skopeo) can be used to inspect the
+contents of the artifact:
+
+```
+> oras pull quay.io/crio/seccomp:v1
+Downloading c15f3eb1efd9 seccomp.tar
+Downloaded  c15f3eb1efd9 seccomp.tar
+Pulled [registry] quay.io/crio/seccomp:v1
+Digest: sha256:c427d4f703263fbd7dbd607bed902eaf405de5880bf89fc344df5fc0a23f4d2a
+
+> tar xf seccomp.tar
+
+> jq . seccomp.json | head
+{
+  "defaultAction": "SCMP_ACT_ERRNO",
+  "defaultErrnoRet": 38,
+  "defaultErrno": "ENOSYS",
+  "archMap": [
+    {
+      "architecture": "SCMP_ARCH_X86_64",
+      "subArchitectures": [
+        "SCMP_ARCH_X86",
+        "SCMP_ARCH_X32"
+```
+
+```
+> skopeo inspect --raw docker://quay.io/crio/seccomp:v1 | jq .
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "digest": "sha256:ca3d163bab055381827226140568f3bef7eaac187cebd76878e0b63e9e442356",
+    "size": 3
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:c15f3eb1efd9abe0ae973e050c374988bba52c45a53f5c3cd80419a865013fa6",
+      "size": 20480,
+      "annotations": {
+        "org.opencontainers.image.title": "seccomp.tar"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2024-02-05T11:37:22Z"
+  }
+}
+```
+
+We can see that the image itself consists of an empty config
```

Comment:
We [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we); there's an exception for using it to refer to the multiple authors of a blog article, but that doesn't apply here.